pkg/controller/planexecution/planexecution_controller.go

```diff
@@ -305,144 +301,55 @@ func (r *ReconcilePlanExecution) Reconcile(request reconcile.Request) (reconcile
-							}
-						} else {
-							log.Printf("PlanExecutionController: Error when applying StrategicMergePatch to object %v for instance %v: %v", key, instance.Name, err)
-						}
-					}
-				} else {
-					//create
-					log.Printf("PlanExecutionController: Object %v does not exist, going to create new object for instance %v", key, instance.Name)
-					err = r.Client.Create(context.TODO(), obj)
-					if err != nil {
-						log.Printf("PlanExecutionController: Error when creating object %v: %v", s.Name, err)
-						planExecution.Status.Phases[i].State = kudov1alpha1.PhaseStateError
-						planExecution.Status.Phases[i].Steps[j].State = kudov1alpha1.PhaseStateError
-
-						return reconcile.Result{}, err
-					}
-				}
-
-				err = health.IsHealthy(r.Client, obj)
-				if err != nil {
-					log.Printf("PlanExecutionController: Obj is NOT healthy: %s", prettyPrint(obj))
-					planExecution.Status.Phases[i].Steps[j].State = kudov1alpha1.PhaseStateInProgress
-					planExecution.Status.Phases[i].State = kudov1alpha1.PhaseStateInProgress
-				}
-			}
-			log.Printf("PlanExecutionController: Phase \"%v\" has strategy %v", phase.Name, phase.Strategy)
-			if phase.Strategy == kudov1alpha1.Serial {
-				// We need to skip the rest of the steps if this step is unhealthy
-				log.Printf("PlanExecutionController: Phase \"%v\" marked as serial", phase.Name)
-				if planExecution.Status.Phases[i].Steps[j].State != kudov1alpha1.PhaseStateComplete {
-					log.Printf("PlanExecutionController: Step \"%v\" isn't complete, skipping rest of steps in phase until it is", planExecution.Status.Phases[i].Steps[j].Name)
-					break
-				} else {
-					log.Printf("PlanExecutionController: Step \"%v\" is healthy, so I can continue on", planExecution.Status.Phases[i].Steps[j].Name)
-				}
-			}
+	if err != nil {
+		log.Printf("PlanExecutionController: error when executing plan for instance %s: %v", instance.Name, err)
 
-			log.Printf("PlanExecutionController: Looked at step \"%v\"", s.Name)
-		}
-		if health.IsPhaseHealthy(planExecution.Status.Phases[i]) {
-			log.Printf("PlanExecutionController: Phase \"%v\" marked as healthy", phase.Name)
-			planExecution.Status.Phases[i].State = kudov1alpha1.PhaseStateComplete
-			continue
+		err = r.Client.Update(context.TODO(), planExecution)
+		if err != nil {
+			log.Printf("PlanExecutionController: Error when updating planExecution state. %v", err)
+			return reconcile.Result{}, err
 		}
 
-		// This phase isn't quite ready yet. Let's see what needs to be done
-		planExecution.Status.Phases[i].State = kudov1alpha1.PhaseStateInProgress
-
-		// Don't keep going to other plans if we're flagged to perform the phases in serial
-		if executedPlan.Strategy == kudov1alpha1.Serial {
-			log.Printf("PlanExecutionController: Phase \"%v\" not healthy, and plan marked as serial, so breaking.", phase.Name)
-			break
+		if _, ok := err.(*fatalError); ok {
```

Comment:
This is a `nil` dereference, because the case that `err != nil` has been handled in the if-clause above.